src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/BlockExchange/BitSwapTest.cs

```diff
@@ -0,0 +1,176 @@
+
+            var cid = new DagNode(Encoding.UTF8.GetBytes("BitswapTest unknown block")).Id;
+            var cancel = new CancellationTokenSource();
+            var _ = bitswap.WantAsync(cid, self.Id, cancel.Token);
+            bitswap.PeerWants(self.Id).ToArray().Should().Contain(cid);
+
+            // Assert.Contains(bitswap.PeerWants(self.Id).ToArray(), cid);
+
+            bitswap.Unwant(cid);
+            bitswap.PeerWants(self.Id).ToArray().Should().NotContain(cid);
+
+            // Assert.DoesNotContain(bitswap.PeerWants(self.Id).ToArray(), cid);
+        }
+
+        [Fact]
+        public void Want_Cancel()
+        {
+            var bitswap = new BitSwapService {SwarmService = new SwarmService() {LocalPeer = self}};
+            var cid = new DagNode(Encoding.UTF8.GetBytes("BitswapTest unknown block")).Id;
+            var cancel = new CancellationTokenSource();
+            var task = bitswap.WantAsync(cid, self.Id, cancel.Token);
+
+            // Assert.Contains(bitswap.PeerWants(self.Id).ToArray(), cid);
+            bitswap.PeerWants(self.Id).ToArray().Should().Contain(cid);
+
+            cancel.Cancel();
+            Assert.True(task.IsCanceled);
+            
+            // Assert.DoesNotContain(bitswap.PeerWants(self.Id).ToArray(), cid);
+
+            bitswap.PeerWants(self.Id).ToArray().Should().NotContain(cid);
+        }
+
+        [Fact]
+        public void Block_Needed()
+        {
+            var bitswap = new BitSwapService {SwarmService = new SwarmService() {LocalPeer = self}};
+            var cid1 = new DagNode(Encoding.UTF8.GetBytes("BitswapTest unknown block y")).Id;
+            var cid2 = new DagNode(Encoding.UTF8.GetBytes("BitswapTest unknown block z")).Id;
+            var cancel = new CancellationTokenSource();
+            var callCount = 0;
+            
+            bitswap.BlockNeeded += (s, e) =>
+            {
+                Assert.True(cid1 == e.Id || cid2 == e.Id);
+                ++callCount;
+            };
+            try
+            {
+                bitswap.WantAsync(cid1, self.Id, cancel.Token);
+                bitswap.WantAsync(cid1, self.Id, cancel.Token);
+                bitswap.WantAsync(cid2, self.Id, cancel.Token);
+                bitswap.WantAsync(cid2, self.Id, cancel.Token);
+                Assert.Equal(2, callCount);
+            }
+            finally
+            {
+                cancel.Cancel();
+            }
+        }
+
+        [Fact]
+        public void Want_Unwant()
+        {
+            var bitswap = new BitSwapService {SwarmService = new SwarmService() {LocalPeer = self}};
+            var cid = new DagNode(Encoding.UTF8.GetBytes("BitswapTest unknown block")).Id;
+            var cancel = new CancellationTokenSource();
+            var task = bitswap.WantAsync(cid, self.Id, cancel.Token);
+            
+            bitswap.PeerWants(self.Id).ToArray().Should().Contain(cid);
+
+            // Assert.Contains(bitswap.PeerWants(self.Id).ToArray(), cid);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)